src/devices/Dhtxx/DhtBase.cs

```diff
@@ -281,6 +281,10 @@ public void Dispose()
             {
                 _controller?.Dispose();
             }
+            else
+            {
+                _controller.ClosePin(_pin);
```

Comment:
Yea, I have couple of others to do ;-) But nothing critical. So as soon as I have to update one, I then correct this :-)

Reply:
Crap, reviewing again suggests this needs to be fixed with a bit more logic. Since calling ClosePin with a pin that is not open throws an exception, this should likely reset the _controller to null, such that a repeated disposal does no harm. Repeatedly disposing the same instance shouldn't throw any exceptions.

Reply:
correct is controller is null. If it's not null, then if the pin is not open, an exception would have been raised much before. so fix is:
```csharp
_controller?.ClosePin(_pin);
```
Good catch!